### PR TITLE
use vBinary.from_ical in Image.data and remove manual base64 decode

### LINKED ISSUE #1550: Add a property for the Base64 string view in vBinary
vBinary is stored as base64 encoded data.
It would be very handy to have `vBinary.base64data` as a property with getter and setter.

- [x] Wait for #1356 to be merged.
- [x] add a getter based on the bytes attribute from #1356.
- [x] add a setter which takes str only and validates the input, raises a ValueError if not base64
- [x] nice docstrings, following our [style guide](https://icalendar.readthedocs.io/en/latest/contribute/documentation/style-guide.html#docstring-structure)
- [x] test all edge cases
- [x] replace all manual base64 encode/decode behaviour in our code base where possible.
 such as https://github.com/collective/icalendar/pull/1356#discussion_r3421606404_

### PR DESCRIPTION
**Pull request by klouds27**
Closes #1550.

PR #1356 added `vBinary.base64data` as a getter/setter and moved internal storage to raw bytes. The one remaining manual base64 call in non-test code was `Image.data`, which called `base64.b64decode(self.b64data)` directly.

Changes:

Replaces `base64.b64decode(self.b64data)` in `Image.data` with `vBinary.from_ical(self.b64data)`. This makes validation strict and consistent with the rest of the codebase, matching the 'Not valid base 64 encoding.' error message.

Removes the dead else `base64.b64encode(value)` branch in `from_property_value`. Since `vText` inherits from `str`, `isinstance(value, str)` is always true in that branch and the else was unreachable.

Drops the now-unused `import base64` from `image.py`.

Adds a test confirming `Image.data` raises `ValueError` with the same message as `vBinary.from_ical` for invalid input.

### CODE REVIEW

**Comment by stevepiercy** [comment]
@klouds27 you marked this PR as a draft. Is it ready for review?

File: news/1550.feature
Comment (stevepiercy) [inline news/1550.feature:1]: Never edit another contributor's news fragment. Instead create your own. See https://icalendar.readthedocs.io/en/latest/contribute/index.html#change-log-entry-file-name.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Review by stevepiercy — APPROVED** [review]
Thanks for fixing the news fragment. This LGTM. Let's get a second technical review.

**Comment by angatha** [comment]
Since you want to close the issue, @stevepiercy are the docstrings nice enough? The class docstring does not document the parameters of `__init__`.

I think the "test all edge cases" part is done. and this does the last checkmark.

**Comment by stevepiercy** [comment]
> Since you want to close the issue, @stevepiercy are the docstrings nice enough? The class docstring does not document the parameters of `__init__`.

I added it explicitly to the description in https://github.com/collective/icalendar/issues/1620. I'd prefer to not hold up this PR. The referenced PR did a pretty good job, but at that time, we hadn't settled on the concatenation of new and init into their class.

> I think the "test all edge cases" part is done. and this does the last checkmark.

Yup. Let's merge.

**Comment by stevepiercy** [comment]
Ugh, it appears GitHub is being GitHub yet again. I got an email notification of @angatha's comment, but it doesn't show up anywhere. So weird! I'll post it as a review.

File: src/icalendar/prop/image.py [hunk 1]
```diff
@@ -58,11 +57,7 @@ def from_property_value(cls, value: vUri | vBinary | vText) -> Image:
         elif isinstance(value, vBinary):
             params["b64data"] = value.base64data
         elif value_type == "BINARY":
-            params["b64data"] = (
-                value
-                if isinstance(value, str)
-                else base64.b64encode(value).decode("ascii")
-            )
+            params["b64data"] = str(value)
```
Comment (stevepiercy) [inline src/icalendar/prop/image.py:60]: From @angatha comment in an email notification.

`isinstance(value, str)` should always be false, so removing that part is fine. Can you please explain, why `base64.b64encode(value).decode("ascii")` can be replaced with `str(value)`? `value` should have type `vText` accoding to the type annotation.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: src/icalendar/prop/image.py (hunk 1 above)
Comment (angatha) [reply src/icalendar/prop/image.py:60]: I removed it, becaute it was wrong. `vText` subclasses `str`.

**Review by stevepiercy — APPROVED** [review]

**Review by angatha — APPROVED** [review]
